linux/deploy_vm/deploy_vm_from_iso.yml

```diff
@@ -65,8 +65,26 @@
     - name: "Compose VM CDROMs to mount OS install ISO files"
       include_tasks: ../../common/compose_vm_cdroms.yml
 
+    # Failed to install FreeBSD 32bit with lsilogic/lsilogicsas/nvme when memory >= 4G
+    - name: "Initialize the fact of memory size when create VM"
+      ansible.builtin.set_fact:
+        vm_memory_mb: |-
+          {%- if guest_id is match('freebsd\\d+Guest') and boot_disk_controller in ['lsilogic', 'lsilogicsas', 'nvme'] -%}
+              {%- if memory_mb is defined and memory_mb and memory_mb | int < 4096 -%}{{ memory_mb }}
+              {%- else -%}3072
+              {%- endif -%}
+          {%- elif memory_mb is defined and memory_mb -%}{{ memory_mb }}
+          {%- else -%}4096
+          {%- endif -%}
```

Comment:
The match pattern is not updated. This expression looks like redundant. You can simplify it as:
````
{%- if guest_id xxxx -%}
{{ [3072, memory_mb | default(4096) | int ] | min }}
{%- else -%}
{{ memory_mb | default(4096) | int  }}
{%- endif -%}
````